applications/serial_lte_modem/src/main.c

```diff
@@ -424,6 +417,9 @@ int main(void)
 			fota_info = -EAGAIN;
 		}
 		return start_execute();
+	} else if (err) {
```

Comment:
@junqingzou: Should we always fail at this point?

Reply:
Must fail and reboot, also I don't think "else if (err)" be handled here, should be right after calling nrf_modem_lib_init() and before the checking FOTA result.